src/Adapter/Customer/Group/CommandHandler/AddCustomerGroupHandler.php

```diff
@@ -0,0 +1,33 @@
+<?php
+
+namespace PrestaShop\PrestaShop\Adapter\Customer\Group\CommandHandler;
+
+use PrestaShop\PrestaShop\Adapter\Customer\Group\Repository\GroupRepository;
+use PrestaShop\PrestaShop\Core\Domain\Customer\Group\Command\AddCustomerGroupCommand;
+use PrestaShop\PrestaShop\Core\Domain\Customer\Group\CommandHandler\AddCustomerGroupHandlerInterface;
+use PrestaShop\PrestaShop\Core\Domain\Customer\Group\ValueObject\GroupId;
+
+class AddCustomerGroupHandler implements AddCustomerGroupHandlerInterface
+{
+    /**
+     * @var GroupRepository
+     */
+    private $customerGroupRepository;
+
+    public function __construct(GroupRepository $customerGroupRepository)
+    {
+        $this->customerGroupRepository = $customerGroupRepository;
+    }
+
+    public function handle(AddCustomerGroupCommand $command): GroupId
+    {
+        $customerGroup = $this->customerGroupRepository->create(
```

Comment:
there is group_shop table which contains shop association. Your command is missing it.
Try to turn on multishop feature and check groups legacy form, you will see shop association selection field, I bet you missed it :stuck_out_tongue_closed_eyes: 
![image](https://user-images.githubusercontent.com/31609858/224303407-49217a9e-b03d-4e78-8157-f114c752369e.png)

Reply:
Nice catch 👍